# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1383593539 18000
#      Mon Nov 04 14:32:19 2013 -0500
# Node ID 3a0075793fcda3164acb8ceb462707b554097ce0
# Parent  42453dcfa95e390e8def515f4fd624cc9a760eec
allow format bank to work for mixed +/- values

* pr-output.cc (set_real_format, set_real_matrix_format,
set_complex_format, set complex_matrix_format):
Set minimum field width to 5.

diff --git a/libinterp/corefcn/pr-output.cc b/libinterp/corefcn/pr-output.cc
--- a/libinterp/corefcn/pr-output.cc
+++ b/libinterp/corefcn/pr-output.cc
@@ -548,19 +548,19 @@ set_real_format (int digits, bool inf_or
 
   if (rat_format)
     {
       fw = 0;
       rd = 0;
     }
   else if (bank_format)
     {
-      fw = digits < 0 ? 4 : digits + 3;
-      if (inf_or_nan && fw < 4)
-        fw = 4;
+      fw = digits < 0 ? 5 : digits + 4;
+      if (inf_or_nan && fw < 5)
+        fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       fw = 2 * sizeof (double);
       rd = 0;
     }
   else if (bit_format)
@@ -674,19 +674,19 @@ set_real_matrix_format (int x_max, int x
   if (rat_format)
     {
       fw = 9;
       rd = 0;
     }
   else if (bank_format)
     {
       int digits = x_max > x_min ? x_max : x_min;
-      fw = digits <= 0 ? 4 : digits + 3;
-      if (inf_or_nan && fw < 4)
-        fw = 4;
+      fw = digits <= 0 ? 5 : digits + 4;
+      if (inf_or_nan && fw < 5)
+        fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       fw = 2 * sizeof (double);
       rd = 0;
     }
   else if (bit_format)
@@ -838,19 +838,19 @@ set_complex_format (int x_max, int x_min
       i_fw = 0;
       r_fw = 0;
       rd = 0;
     }
   else if (bank_format)
     {
       int digits = r_x;
       i_fw = 0;
-      r_fw = digits <= 0 ? 4 : digits + 3;
-      if (inf_or_nan && r_fw < 4)
-        r_fw = 4;
+      r_fw = digits <= 0 ? 5 : digits + 4;
+      if (inf_or_nan && r_fw < 5)
+        r_fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       r_fw = 2 * sizeof (double);
       i_fw = 2 * sizeof (double);
       rd = 0;
     }
@@ -1038,19 +1038,19 @@ set_complex_matrix_format (int x_max, in
       i_fw = 9;
       r_fw = 9;
       rd = 0;
     }
   else if (bank_format)
     {
       int digits = r_x_max > r_x_min ? r_x_max : r_x_min;
       i_fw = 0;
-      r_fw = digits <= 0 ? 4 : digits + 3;
-      if (inf_or_nan && r_fw < 4)
-        r_fw = 4;
+      r_fw = digits <= 0 ? 5 : digits + 4;
+      if (inf_or_nan && r_fw < 5)
+        r_fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       r_fw = 2 * sizeof (double);
       i_fw = 2 * sizeof (double);
       rd = 0;
     }
